cmd/dependabot/internal/cmd/update.go

```diff
@@ -184,11 +185,12 @@ func init() {
 	updateCmd.Flags().StringVarP(&file, "file", "f", "", "path to scenario file")
 
 	updateCmd.Flags().StringVarP(&provider, "provider", "p", "github", "provider of the repository")
-	updateCmd.Flags().StringVarP(&directory, "directory", "d", "/", "directory to update")
+	updateCmd.Flags().StringVarP(&directory, "directory", "d", "/", "directory in the repo to update")
 
 	updateCmd.Flags().BoolVar(&dryRun, "dry-run", true, "perform update as a dry run")
 	_ = updateCmd.MarkFlagRequired("dry-run")
 
+	updateCmd.Flags().StringVar(&localRepo, "local-repo", "", "local repo to update, skips fetch_files step")
```

Comment:
Instead of adding a new CLI option, could we maybe repurpose the ones we have already?

One idea is that we could define a new `local` provider and use the `directory` option to point to a directory in a local Git repo (so if `-d repo/nested/subdirectory` then we recursively check parent directories until we find one containing a `.git` directory).

Reply:
That's an interesting thought @mattt.

I know there's a few places we explode on unhandled providers like [here](https://github.com/dependabot/dependabot-core/blob/b7c25bc8c776d6571b63b645a89871f5d0b8501b/git_submodules/lib/dependabot/git_submodules/file_fetcher.rb#L57). So we'd need to fix that up. 

I'll ponder this more and see what other folks think. This feature isn't a blocker so no rush.

Reply:
That's a good callout about core raising exceptions for unsupported providers. I don't have a sense of how much of a lift that'd be to fix that in core, or at least whether this feature can justify that work. (Long-term, I'd like for us to move from "repo / provider" to normalized Git URLs, which are more flexible and correctly handle things like GHES, but I digress...)

I'm not a fan of [Docker `--mount` syntax](https://docs.docker.com/storage/volumes/#choose-the--v-or---mount-flag), but if we can easily do a 1-1 translation using Moby, then maybe that'd be a reasonable option for doing a read-only mount of source. Curious to hear what everyone else thinks about that.

Reply:
I think the semantics of setting `provider: local` is a little different than what I'm trying to achieve here. For testing purposes only I want to mount the `/home/dependabot/dependabot-updater/repo` directory so that I don't have to push changes to a remote source in order to try it in a dry-run. Everything else I want to stay the same, still take into account the original provider, still post PRs, etc.

To set `provider: local` I am telling the updater I want to update the local filesystem, and the CLI could figure out what to mount from that. When it comes time to post a PR it could update the local filesystem

That sounds like a cool feature but not what I'm going for here. 

However I realized I already implemented volume support with `-v` and so we can already do what I was going for here! So we can close this. Thanks for the brake check @mattt!